classes/Hotel.js

```diff
@@ -24,9 +25,56 @@ class Hotel {
     return bookings ? bookings.map(booking => new Booking(booking)) : [];
   }
 
-  // filterRooms(type, value) {
-  //   this.
-  // }
+  filterRooms(type, value) {
+    this.filteredRooms = this.filteredRooms.filter(room => {
+      return room[type] === value
+    })
+  }
+
+  filterByDate(date) {
+    const convertedDate = date.split('-').join('/');
+    const bookings = this.bookings.filter(booking => {
+      return booking.date === convertedDate
+    });
+    const roomNumbers = bookings.map(booking => booking.roomNumber);
+    this.filteredRooms = this.rooms.filter(room => {
+      return !roomNumbers.includes(room.number)
+    })
+  }
+
+  bookRoom(info) {
+    this.bookings.push(new Booking(info));
+  }
+
+  cancelBooking(id) {
+    const booking = this.bookings.find(book => book.id === id);
+    const index = this.bookings.indexOf(booking);
+    this.bookings.splice(index, 1);
+  }
```

Comment:
Nice use of 'indexOf' here to cancel a booking. It shows you're considering a comprehensive user experience.